sphinx_gallery/backreferences.py

```diff
@@ -341,12 +366,12 @@ def _thumbnail_div(
 
     template = BACKREF_THUMBNAIL_TEMPLATE if is_backref else THUMBNAIL_TEMPLATE
     return template.format(
-        snippet=escape(snippet), thumbnail=thumb, title=title, ref_name=ref_name
+        intro=escape(intro), thumbnail=thumb, title=title, ref_name=ref_name
     )
 
 
 def _write_backreferences(
```

Comment:
This now also returns a dict of backreferences. Thought about changing the function name but maybe just updating the docstring is adequate?

Reply:
I think it's worth changing it even though it's some code churn. A `_write_` function not writing something will be confusing later I think

Reply:
It does write, the `.examples` files are still being written, exactly as before. I think we needed to keep these files to be backwards compatible, and I guess it isn't a huge expense at build time.

I kept the name as I think it is not uncommon for functions to both write out and return, and `_write_and_return_backreferences` is a bit unwieldy.